sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuShuffledHashJoinExec.scala

```diff
@@ -314,6 +329,78 @@ object GpuShuffledHashJoinExec {
     }
   }
 
+  private def getBuildIterMaybeFiltered(
+      gpuBuildIter: Iterator[ColumnarBatch],
+      buildGoal: CoalesceSizeGoal): (Iterator[ColumnarBatch], Boolean) = {
+    var accBatchRows = 0L
```

Comment:
We saw performance issues no matter how many rows there were. I think we want to filter the input data any time that the goal is RequireSingleBatchWithFilter.

And if that is the case, I don't see a lot of reason to keep this as a separate method.

Reply:
Actually I did as your suggestion here (you can find it in [the first commit ](https://github.com/NVIDIA/spark-rapids/pull/8659/commits/3a22e1ef2eb560951900cfd2786f080c90fa4027)in this PR), but it led to more than 300 failing tests. (For more you can refer to premerge build NO.7443).

If in theory it is safe to filter out nulls for these special join types regardless of the rows number, we can have follow-ups.

```
[2023-07-05T08:34:53.178Z] 2023-07-05 07:41:54 INFO     Skipping findspark init because on xdist master

[2023-07-05T08:34:53.178Z] FAILED ../../src/main/python/join_test.py::test_struct_self_join[INJECT_OOM, IGNORE_ORDER({'local': True})] - py4j.protocol.Py4JJavaError: An error occurred while calling o663261.collectToPython.

[2023-07-05T08:34:53.178Z] : org.apache.spark.SparkException: Job aborted due to stage failure: Task 1 in stage 11905.0 failed 1 times, most recent failure: Lost task 1.0 in stage 11905.0 (TID 63708) (premerge-ci-2-jenkins-rapids-premerge-github-7443-r0pz3-15h93 executor driver): ai.rapids.cudf.CudfException: CUDF failure at:/home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-475-cuda11/thirdparty/cudf/cpp/src/unary/cast_ops.cu:393: Column type must be numeric or chrono or decimal32/64/128

[2023-07-05T08:34:53.179Z] 	at ai.rapids.cudf.ColumnView.castTo(Native Method)

[2023-07-05T08:34:53.179Z] 	at ai.rapids.cudf.ColumnView.castTo(ColumnView.java:1946)

[2023-07-05T08:34:53.179Z] 	at ai.rapids.cudf.ColumnVector.castTo(ColumnVector.java:836)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuCast$.doCast(GpuCast.scala:531)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuCast.doColumnar(GpuCast.scala:1567)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuUnaryExpression.doItColumnar(GpuExpressions.scala:194)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuUnaryExpression.$anonfun$columnarEval$1(GpuExpressions.scala:205)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:29)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuUnaryExpression.columnarEval(GpuExpressions.scala:204)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.RapidsPluginImplicits$ReallyAGpuExpression.columnarEval(implicits.scala:34)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuCoalesce.$anonfun$columnarEval$1(nullExpressions.scala:54)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuCoalesce.$anonfun$columnarEval$1$adapted(nullExpressions.scala:53)

[2023-07-05T08:34:53.179Z] 	at scala.collection.immutable.List.foreach(List.scala:392)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuCoalesce.columnarEval(nullExpressions.scala:53)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.RapidsPluginImplicits$ReallyAGpuExpression.columnarEval(implicits.scala:34)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuExpressionsUtils$.columnarEvalToColumn(GpuExpressions.scala:95)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuProjectExec$.projectSingle(basicPhysicalOperators.scala:104)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuFilter$.computeCheckedFilterMask(basicPhysicalOperators.scala:699)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuFilter$.apply(basicPhysicalOperators.scala:724)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuShuffledHashJoinExec$.$anonfun$filterBatchAndClose$1(GpuShuffledHashJoinExec.scala:375)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:29)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuShuffledHashJoinExec$.filterBatchAndClose(GpuShuffledHashJoinExec.scala:375)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuShuffledHashJoinExec$.$anonfun$getBuildBatchOptimizedAndClose$4(GpuShuffledHashJoinExec.scala:412)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuMetric.ns(GpuExec.scala:150)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuShuffledHashJoinExec$.$anonfun$getBuildBatchOptimizedAndClose$2(GpuShuffledHashJoinExec.scala:409)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.Arm$.closeOnExcept(Arm.scala:88)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuShuffledHashJoinExec$.$anonfun$getBuildBatchOptimizedAndClose$1(GpuShuffledHashJoinExec.scala:400)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:29)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuShuffledHashJoinExec$.getBuildBatchOptimizedAndClose(GpuShuffledHashJoinExec.scala:399)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuShuffledHashJoinExec$.$anonfun$prepareBuildBatchesForJoin$2(GpuShuffledHashJoinExec.scala:287)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.Arm$.closeOnExcept(Arm.scala:88)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuShuffledHashJoinExec$.prepareBuildBatchesForJoin(GpuShuffledHashJoinExec.scala:260)

[2023-07-05T08:34:53.179Z] 	at com.nvidia.spark.rapids.GpuShuffledHashJoinExec.$anonfun$internalDoExecuteColumnar$2(GpuShuffledHashJoinExec.scala:185)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.rdd.ZippedPartitionsRDD2.compute(ZippedPartitionsRDD.scala:89)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.scheduler.Task.run(Task.scala:131)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:497)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1439)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:500)

[2023-07-05T08:34:53.179Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)

[2023-07-05T08:34:53.179Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)

[2023-07-05T08:34:53.179Z] 	at java.lang.Thread.run(Thread.java:750)

[2023-07-05T08:34:53.179Z] 

[2023-07-05T08:34:53.179Z] Driver stacktrace:

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2253)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2202)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2201)

[2023-07-05T08:34:53.179Z] 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)

[2023-07-05T08:34:53.179Z] 	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)

[2023-07-05T08:34:53.179Z] 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2201)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1078)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1078)

[2023-07-05T08:34:53.179Z] 	at scala.Option.foreach(Option.scala:407)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1078)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2440)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2382)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2371)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:868)

[2023-07-05T08:34:53.179Z] 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2202)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2223)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2242)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2267)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.RDD.$anonfun$collect$1(RDD.scala:1030)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.RDD.withScope(RDD.scala:414)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.RDD.collect(RDD.scala:1029)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.sql.execution.SparkPlan.executeCollect(SparkPlan.scala:390)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.sql.Dataset.$anonfun$collectToPython$1(Dataset.scala:3519)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:3687)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$5(SQLExecution.scala:103)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:163)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:90)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:772)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3685)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.sql.Dataset.collectToPython(Dataset.scala:3516)

[2023-07-05T08:34:53.180Z] 	at sun.reflect.GeneratedMethodAccessor120.invoke(Unknown Source)

[2023-07-05T08:34:53.180Z] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)

[2023-07-05T08:34:53.180Z] 	at java.lang.reflect.Method.invoke(Method.java:498)

[2023-07-05T08:34:53.180Z] 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)

[2023-07-05T08:34:53.180Z] 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)

[2023-07-05T08:34:53.180Z] 	at py4j.Gateway.invoke(Gateway.java:282)

[2023-07-05T08:34:53.180Z] 	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)

[2023-07-05T08:34:53.180Z] 	at py4j.commands.CallCommand.execute(CallCommand.java:79)

[2023-07-05T08:34:53.180Z] 	at py4j.GatewayConnection.run(GatewayConnection.java:238)

[2023-07-05T08:34:53.180Z] 	at java.lang.Thread.run(Thread.java:750)

[2023-07-05T08:34:53.180Z] Caused by: ai.rapids.cudf.CudfException: CUDF failure at:/home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-475-cuda11/thirdparty/cudf/cpp/src/unary/cast_ops.cu:393: Column type must be numeric or chrono or decimal32/64/128

[2023-07-05T08:34:53.180Z] 	at ai.rapids.cudf.ColumnView.castTo(Native Method)

[2023-07-05T08:34:53.180Z] 	at ai.rapids.cudf.ColumnView.castTo(ColumnView.java:1946)

[2023-07-05T08:34:53.180Z] 	at ai.rapids.cudf.ColumnVector.castTo(ColumnVector.java:836)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuCast$.doCast(GpuCast.scala:531)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuCast.doColumnar(GpuCast.scala:1567)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuUnaryExpression.doItColumnar(GpuExpressions.scala:194)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuUnaryExpression.$anonfun$columnarEval$1(GpuExpressions.scala:205)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:29)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuUnaryExpression.columnarEval(GpuExpressions.scala:204)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.RapidsPluginImplicits$ReallyAGpuExpression.columnarEval(implicits.scala:34)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuCoalesce.$anonfun$columnarEval$1(nullExpressions.scala:54)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuCoalesce.$anonfun$columnarEval$1$adapted(nullExpressions.scala:53)

[2023-07-05T08:34:53.180Z] 	at scala.collection.immutable.List.foreach(List.scala:392)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuCoalesce.columnarEval(nullExpressions.scala:53)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.RapidsPluginImplicits$ReallyAGpuExpression.columnarEval(implicits.scala:34)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuExpressionsUtils$.columnarEvalToColumn(GpuExpressions.scala:95)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuProjectExec$.projectSingle(basicPhysicalOperators.scala:104)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuFilter$.computeCheckedFilterMask(basicPhysicalOperators.scala:699)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuFilter$.apply(basicPhysicalOperators.scala:724)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuShuffledHashJoinExec$.$anonfun$filterBatchAndClose$1(GpuShuffledHashJoinExec.scala:375)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:29)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuShuffledHashJoinExec$.filterBatchAndClose(GpuShuffledHashJoinExec.scala:375)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuShuffledHashJoinExec$.$anonfun$getBuildBatchOptimizedAndClose$4(GpuShuffledHashJoinExec.scala:412)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuMetric.ns(GpuExec.scala:150)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuShuffledHashJoinExec$.$anonfun$getBuildBatchOptimizedAndClose$2(GpuShuffledHashJoinExec.scala:409)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.Arm$.closeOnExcept(Arm.scala:88)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuShuffledHashJoinExec$.$anonfun$getBuildBatchOptimizedAndClose$1(GpuShuffledHashJoinExec.scala:400)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:29)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuShuffledHashJoinExec$.getBuildBatchOptimizedAndClose(GpuShuffledHashJoinExec.scala:399)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuShuffledHashJoinExec$.$anonfun$prepareBuildBatchesForJoin$2(GpuShuffledHashJoinExec.scala:287)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.Arm$.closeOnExcept(Arm.scala:88)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuShuffledHashJoinExec$.prepareBuildBatchesForJoin(GpuShuffledHashJoinExec.scala:260)

[2023-07-05T08:34:53.180Z] 	at com.nvidia.spark.rapids.GpuShuffledHashJoinExec.$anonfun$internalDoExecuteColumnar$2(GpuShuffledHashJoinExec.scala:185)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.ZippedPartitionsRDD2.compute(ZippedPartitionsRDD.scala:89)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)

[2023-07-05T08:34:53.180Z] 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)

[2023-07-05T08:34:53.181Z] 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)

[2023-07-05T08:34:53.181Z] 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)

[2023-07-05T08:34:53.181Z] 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)

[2023-07-05T08:34:53.181Z] 	at org.apache.spark.scheduler.Task.run(Task.scala:131)

[2023-07-05T08:34:53.181Z] 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:497)

[2023-07-05T08:34:53.181Z] 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1439)

[2023-07-05T08:34:53.181Z] 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:500)

[2023-07-05T08:34:53.181Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)

[2023-07-05T08:34:53.181Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)

[2023-07-05T08:34:53.181Z] 	... 1 more

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_multi_table_hash_join[True-True-String][INJECT_OOM, IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 47 GPU: 0

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_multi_table_hash_join[True-False-String][IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 47 GPU: 0

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_multi_table_hash_join[False-True-String][IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 47 GPU: 0

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_multi_table_hash_join[False-False-String][INJECT_OOM, IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 47 GPU: 0

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_hash_join_different_key_integral_types[Inner-Byte-Byte][INJECT_OOM, IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 125 GPU: 123

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_hash_join_different_key_integral_types[Inner-Short-Byte][IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 2 GPU: 1

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_hash_join_different_key_integral_types[Inner-Integer-Byte][IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 1 GPU: 0

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_hash_join_different_key_integral_types[Inner-Long-Byte][IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 3 GPU: 0

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_hash_join_different_key_integral_types[LeftSemi-Byte-Byte][IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 46 GPU: 45

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_hash_join_different_key_integral_types[LeftSemi-Short-Byte][IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 2 GPU: 1

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_hash_join_different_key_integral_types[LeftSemi-Integer-Byte][INJECT_OOM, IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 1 GPU: 0

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_hash_join_different_key_integral_types[LeftSemi-Long-Byte][IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 1 GPU: 0

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_hash_join_different_key_integral_types[LeftAnti-Byte-Byte][IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 4 GPU: 5

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_hash_join_different_key_integral_types[LeftAnti-Short-Byte][IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 48 GPU: 49

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_hash_join_different_key_integral_types[LeftAnti-Integer-Byte][INJECT_OOM, IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 49 GPU: 50

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_hash_join_different_key_integral_types[LeftAnti-Long-Byte][IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 49 GPU: 50

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_hash_join_different_key_integral_types[Cross-Byte-Byte][IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 125 GPU: 123

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_hash_join_different_key_integral_types[Cross-Short-Byte][INJECT_OOM, IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 2 GPU: 1

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_hash_join_different_key_integral_types[Cross-Integer-Byte][IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 1 GPU: 0

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/join_test.py::test_hash_join_different_key_integral_types[Cross-Long-Byte][INJECT_OOM, IGNORE_ORDER({'local': True})] - AssertionError: CPU and GPU list have different lengths at [] CPU: 3 GPU: 0

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[-reader_confs0][IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[-reader_confs1][IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[-reader_confs2][IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[-reader_confs3][IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[-reader_confs4][INJECT_OOM, IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.181Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[-reader_confs5][IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[-reader_confs6][IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[-reader_confs7][INJECT_OOM, IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[-reader_confs8][INJECT_OOM, IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[-reader_confs9][INJECT_OOM, IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[-reader_confs10][INJECT_OOM, IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[parquet-reader_confs0][IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[parquet-reader_confs1][IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[parquet-reader_confs2][IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[parquet-reader_confs3][IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/qa_nightly_select_test.py::test_needs_sort_select[A.shortF, B.intF A, test_table B where A.intF=B.intF ORDER BY A.intF][IGNORE_ORDER, INCOMPAT, APPROXIMATE_FLOAT] - AssertionError: CPU and GPU list have different lengths at [] CPU: 37 GPU: 36

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/qa_nightly_select_test.py::test_needs_sort_select[A.shortF, B.intF A, test_table B where A.intF=B.intF][INJECT_OOM, IGNORE_ORDER, INCOMPAT, APPROXIMATE_FLOAT] - AssertionError: CPU and GPU list have different lengths at [] CPU: 37 GPU: 36

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/qa_nightly_select_test.py::test_select_join[A.byteF, B.byteF A, test_table B where A.byteF=B.byteF][INJECT_OOM, IGNORE_ORDER({'local': True}), INCOMPAT, APPROXIMATE_FLOAT] - AssertionError: CPU and GPU list have different lengths at [] CPU: 50 GPU: 41

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/qa_nightly_select_test.py::test_select_join[A.shortF, B.shortF A, test_table B where A.shortF=B.shortF][INJECT_OOM, IGNORE_ORDER({'local': True}), INCOMPAT, APPROXIMATE_FLOAT] - AssertionError: CPU and GPU list have different lengths at [] CPU: 49 GPU: 48

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/qa_nightly_select_test.py::test_select_join[A.intF, B.intF A, test_table B where A.intF=B.intF][INJECT_OOM, IGNORE_ORDER({'local': True}), INCOMPAT, APPROXIMATE_FLOAT] - AssertionError: CPU and GPU list have different lengths at [] CPU: 37 GPU: 36

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/qa_nightly_select_test.py::test_select_join[A.longF, B.longF A, test_table B where A.longF=B.longF][IGNORE_ORDER({'local': True}), INCOMPAT, APPROXIMATE_FLOAT] - AssertionError: CPU and GPU list have different lengths at [] CPU: 48 GPU: 47

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/qa_nightly_select_test.py::test_select_join[A.floatF, B.floatF A, test_table B where A.floatF=B.floatF][IGNORE_ORDER({'local': True}), INCOMPAT, APPROXIMATE_FLOAT] - AssertionError: CPU and GPU list have different lengths at [] CPU: 40 GPU: 39

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/qa_nightly_select_test.py::test_select_join[A.doubleF, B.doubleF A, test_table B where A.doubleF=B.doubleF][INJECT_OOM, IGNORE_ORDER({'local': True}), INCOMPAT, APPROXIMATE_FLOAT] - AssertionError: CPU and GPU list have different lengths at [] CPU: 52 GPU: 51

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/qa_nightly_select_test.py::test_select_join[A.booleanF, B.booleanF A, test_table B where A.booleanF=B.booleanF][INJECT_OOM, IGNORE_ORDER({'local': True}), INCOMPAT, APPROXIMATE_FLOAT] - AssertionError: CPU and GPU list have different lengths at [] CPU: 116 GPU: 16

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[parquet-reader_confs4][INJECT_OOM, IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/qa_nightly_select_test.py::test_select_join[A.strF, B.strF A, test_table B where A.strF=B.strF][IGNORE_ORDER({'local': True}), INCOMPAT, APPROXIMATE_FLOAT] - AssertionError: CPU and GPU list have different lengths at [] CPU: 19 GPU: 0

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/qa_nightly_select_test.py::test_select_join[A.dateF, B.dateF A, test_table B where A.dateF=B.dateF][INJECT_OOM, IGNORE_ORDER({'local': True}), INCOMPAT, APPROXIMATE_FLOAT] - AssertionError: CPU and GPU list have different lengths at [] CPU: 196 GPU: 0

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/qa_nightly_select_test.py::test_select_join[test_table.booleanF, test_table1.booleanF INNER JOIN test_table1 ON test_table.booleanF=test_table1.booleanF][INJECT_OOM, IGNORE_ORDER({'local': True}), INCOMPAT, APPROXIMATE_FLOAT] - AssertionError: CPU and GPU list have different lengths at [] CPU: 72 GPU: 12

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/qa_nightly_select_test.py::test_select_join[test_table.strF, test_table1.strF INNER JOIN test_table1 ON test_table.strF=test_table1.strF][IGNORE_ORDER({'local': True}), INCOMPAT, APPROXIMATE_FLOAT] - AssertionError: CPU and GPU list have different lengths at [] CPU: 3 GPU: 0

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[parquet-reader_confs5][IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[parquet-reader_confs6][IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.182Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[parquet-reader_confs7][IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.183Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[parquet-reader_confs8][IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.183Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[parquet-reader_confs9][IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.183Z] FAILED ../../src/main/python/parquet_test.py::test_buckets[parquet-reader_confs10][IGNORE_ORDER, ALLOW_NON_GPU(DataWritingCommandExec,ExecutedCommandExec,WriteFilesExec)] - AssertionError: CPU and GPU list have different lengths at [] CPU: 100000 GPU: 99999

[2023-07-05T08:34:53.183Z] = 385 failed, 17427 passed, 1357 skipped, 637 xfailed, 268 xpassed, 69 warnings in 3163.37s (0:52:43) =
```

Reply:
Then how can we ship this? The size of the data should have no impact on if we should remove nulls from it or not.  All it is doing is turning this off for tests.

Reply:
> The size of the data should have no impact on if we should remove nulls from it or not.

Thx and good to know. I applied your change and it works. Then updated the PR accordingly.